comp/core/flare/flare.go

```diff
@@ -188,14 +188,45 @@ func (f *flare) Create(pdata ProfileData, ipcError error) (string, error) {
 		f.collectConfigFiles,
 	)
 
+	f.runProviders(fb, providers)
+
+	return fb.Save()
+}
+
+func (f *flare) runProviders(fb types.FlareBuilder, providers []types.FlareCallback) {
+	flareStepTimeout := time.Duration(f.config.GetInt("flare_provider_timeout")) * time.Second
+	timer := time.NewTimer(flareStepTimeout)
+	defer timer.Stop()
+
 	for _, p := range providers {
-		err = p(fb)
-		if err != nil {
-			f.log.Errorf("error calling '%s' for flare creation: %s",
-				runtime.FuncForPC(reflect.ValueOf(p).Pointer()).Name(), // reflect p.Callback function name
-				err)
+		providerName := runtime.FuncForPC(reflect.ValueOf(p).Pointer()).Name()
+		f.log.Infof("Running flare provider %s", providerName)
+
+		done := make(chan struct{})
+		go func() {
+			startTime := time.Now()
+			err := p(fb)
+			duration := time.Since(startTime)
+
+			if err == nil {
+				f.log.Debugf("flare provider '%s' completed in %s", providerName, duration)
```

Comment:
We don't need to, but I think it can be helpful to debug issues where the flare command blocks (or now that it can't block anymore, issues where some provider takes too long).
I was also thinking we could add telemetry around that to know what provider tends to take too long.

If you feel like having the actual duration is useless then I'll remove it, let me know